### PR TITLE
Add DataFrames.jl GSOC proposal to tables.md

### PR DESCRIPTION
**Pull request by bkamins**
CC @logankilpatrick

### CODE REVIEW

File: jsoc/gsoc/tables.md [hunk 1]
```diff
@@ -48,4 +48,37 @@
...
\ No newline at end of file
+**Mentors:** [Tanmay Mohapatra](https://github.com/tanmaykm)
+
+## DataFrames.jl join enhancements
+
+**Difficulty**: Hard
+
+**Duration**: 175 hours 
+
+[DataFrames.jl](https://github.com/JuliaData/DataFrames.jl) is one of the more popular implementations of tabular data type for Julia. One of the features it supports is data frame joining. However, more work is needed to improve this functionality. The specific targets for this project are (a final list of targets included in the scope of the project can be decided later).
```
Comment (logankilpatrick) [inline jsoc/gsoc/tables.md:59]: Hey @bkamins, ideally we would generally scope out these projects, the Google team advises against just including a list of open issues unless they are specific for GSoC project options, can you port over a high level work flow on a few of these?

File: jsoc/gsoc/tables.md [hunk 2]
```diff
@@ -48,4 +48,31 @@
...
+@@tight-list
+* fully implement multi-threading support of joins, reduce memory requirements of used join algorithms (which should additionally improve their performance), verify efficiency of alternative joining strategies and implement them along with adaptive algorithm choosing the right joining strategy depending on the passed data;
```
Comment (bkamins) [inline jsoc/gsoc/tables.md:61]:
```suggestion
* fully implement multi-threading support by joins, reduce memory requirements of used join algorithms (which should additionally improve their performance), verify efficiency of alternative joining strategies and implement them along with adaptive algorithm choosing the right joining strategy depending on the passed data;
```

File: jsoc/gsoc/tables.md [hunk 3]
```diff
@@ -48,4 +48,31 @@
...
+
+## DataFrames.jl join enhancements
+
+**Difficulty**: Hard
+
+**Duration**: 175 hours 
+
+[DataFrames.jl](https://github.com/JuliaData/DataFrames.jl) is one of the more popular implementations of tabular data type for Julia. One of the features it supports is data frame joining. However, more work is needed to improve this functionality. The specific targets for this project are (a final list of targets included in the scope of the project can be decided later).
+@@tight-list
+* fully implement multi-threading support by joins, reduce memory requirements of used join algorithms (which should additionally improve their performance), verify efficiency of alternative joining strategies and implement them along with adaptive algorithm choosing the right joining strategy depending on the passed data;
```
Comment (bkamins) [inline jsoc/gsoc/tables.md:61]:
```suggestion
* fully implement multi-threading support by joins, reduce memory requirements of used join algorithms (which should additionally improve their performance), verify efficiency of alternative joining strategies in comparison to those currently used and implement them along with adaptive algorithm choosing the right joining strategy depending on the passed data;
```

File: jsoc/gsoc/tables.md [hunk 4]
```diff
@@ -48,4 +48,31 @@
...
+## DataFrames.jl join enhancements
+
+**Difficulty**: Hard
+
+**Duration**: 175 hours 
+
+[DataFrames.jl](https://github.com/JuliaData/DataFrames.jl) is one of the more popular implementations of tabular data type for Julia. One of the features it supports is data frame joining. However, more work is needed to improve this functionality. The specific targets for this project are (a final list of targets included in the scope of the project can be decided later).
+@@tight-list
+* fully implement multi-threading support by joins, reduce memory requirements of used join algorithms (which should additionally improve their performance), verify efficiency of alternative joining strategies in comparison to those currently used and implement them along with adaptive algorithm choosing the right joining strategy depending on the passed data;
+* implement join allowing for efficient matching on non-equal keys; special attention should be made to matching on keys that are timestamps and spatial objects;
```
Comment (bkamins) [inline jsoc/gsoc/tables.md:62]:
```suggestion
* implement join allowing for efficient matching on non-equal keys; special attention should be made to matching on keys that are date/time and spatial objects;
```

**Comment by logankilpatrick** [comment]
Yes, thank you!

**Pull request merged**